src/TLuaInterpreter.cpp

```diff
@@ -219,7 +219,11 @@ TLuaInterpreter::~TLuaInterpreter()
 // With reduced repetition like that:
 //    bool showOnTop = getVerifiedBool(L, "createMapLabel", 14, "showOnTop", true);
 //
-// The "isOptional" parameter is optional, and will default to not-optional parameters! :)
+// The "isOptional" parameter is optional but modifies the error message to say
+// that an argument is optional and it will default to not-optional parameters!
+// HOWEVER it does not actually handle the absence of an argument that is
+// supposed to BE optional - that has to be done by the caller before it
+// makes the call... 8-P
```

Comment:
These are *internal* use functions though - only seen by us developers - who will need to remember this detail! :imp:

Reply:
We have non-native English speaking developers (and would love more), can you reword this to be simpler?

Reply:
@vadi2 (@Vadi#3695)
```diff
-// The "isOptional" parameter is optional but modifies the error message to say
-// that an argument is optional and it will default to not-optional parameters!
-// HOWEVER it does not actually handle the absence of an argument that is
-// supposed to BE optional - that has to be done by the caller before it
-// makes the call... 8-P
+// The (optional here) "isOptional" parameter only modifies the error message
+// given to the user to say that the parameter need not be provided. It does not
+// (until someone works out how to!) allow this function to deal with the argument
+// at the given pos not being present - so it is up to the function that calls this to
+// check for that BEFORE trying to call this function. This applies to all the
+// "getVerifiedXxxx" functions.
```

Reply:
Sounds good!